app/src/chainlit.py

```diff
@@ -1,32 +1,61 @@
 import logging
+from urllib.parse import parse_qs, urlparse
 
 import chainlit as cl
-import src.adapters.db as db
-from src.format import format_guru_cards
-from src.generate import generate
+from src import chat_engine
 from src.login import require_login
-from src.retrieve import retrieve
-from src.shared import get_embedding_model
 
 logger = logging.getLogger(__name__)
 
 require_login()
 
 
-@cl.on_message
-async def main(message: cl.Message) -> None:
-    logger.info(f"Received: {message.content!r}")
+@cl.on_chat_start
+async def start() -> None:
+    engine_id = engine_url_query_value()
+    logger.info("engine_id: %s", engine_id)
+    engine = chat_engine.create_engine(engine_id)
+    if not engine:
+        await cl.Message(
+            author="backend",
+            metadata={"engine": engine_id},
+            content=f"Available engines: {chat_engine.available_engines()}",
+        ).send()
+        return
+
+    cl.user_session.set("chat_engine", engine)
+    await engine.on_start()
+    await cl.Message(
+        author="backend",
+        metadata={"engine": engine_id},
+        content=f"Chat engine started: {engine.name}",
+    ).send()
 
-    with db.PostgresDBClient().get_session() as db_session:
-        chunks = retrieve(
-            db_session,
-            get_embedding_model(),
-            message.content,
-        )
 
-    response = generate(message.content, context=chunks)
-    formatted_guru_cards = format_guru_cards(chunks)
+def engine_url_query_value() -> str:
+    url = cl.user_session.get("http_referer")
+    logger.debug("URL: %s", url)
```

Comment:
```suggestion
    logger.debug("Referer URL: %s", url)
```
(keeping the logs slightly more precise)